public/app/plugins/datasource/cloudwatch/utils/datalinks.ts

```diff
@@ -16,7 +16,7 @@ export async function addDataLinksToLogsResponse(
   request: DataQueryRequest<CloudWatchQuery>,
   range: TimeRange,
   replaceFn: ReplaceFn,
-  getVariableValueFn: (value: string, scopedVars: ScopedVars) => string[],
+  getVariableValueFn: (value: string, scopedVars: ScopedVars) => string | string[],
```

Comment:
Don't think you need to make any changes in this file.

Reply:
This `addDataLinksToLogsResponse` is called [here](https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/cloudwatch/query-runner/CloudWatchLogsQueryRunner.ts#L153) using `expandVariableToArray` as its `getVariableValueFn`. So changing the return signature of `expandVariableToArray` means we need to change it here too

Reply:
Okay I see!